### ISSUE TITLE
Add Resource Quotas

### ISSUE THREAD

**Issue by tot19** [hvac/hvac#1071]
Hello all, thank you so much for your awesome contributions; this project is really great!

Was wondering if there was any interest to add resource quotas to hvac? [Here](https://developer.hashicorp.com/vault/docs/concepts/resource-quotas) is the HashiCorp doc on that.

Would be happy to submit the PR, but I want to ensure that the feature is desired by broader community. Thanks!

**Comment by Tylerlhess** [comment]
I don't see why it would hurt to implement it. I had some coworkers (vault admins) that used it as a password store with an auto lookup script that constantly traversed all the secrets. They were single handedly using 1/3 of the calls each. I think this could have been reduced if they used vault agent locally for some caching but just thought of that. 

TL;DR; yes make the PR. Someone will find it useful to have implemented.

**Comment by emoryotott19** [comment]
Thanks for the response! I will get started on that
